elephant/causality/granger.py

```diff
@@ -0,0 +1,382 @@
+                               block_row*dimension:
+                               (block_row+1)*dimension] = \
+                lag_covariances[block_column-block_row]
+    return yule_walker_matrix, lag_covariances
+
+
+def _vector_arm(signals, dimension, order):
+    r"""
+    Determine coefficients of autoregressive model from time series data
+    Parameters
+    ----------
+    signals : np.ndarray
+        time series data
+    order : int
+        order of the autoregressive model
+    Returns
+    -------
+    coeffs: np.ndarray
+        coefficients of the autoregressive model
+        ry
+    covar_mat : np.ndarray
+        covariance matrix of
+
+    Coefficients of autoregressive model calculated via solving the linear
+    equation
+
+    M A = C
+
+    where
+    M: Yule-Waler Matrix
+    A: Coefficients of autoregressive model
+    C: Time-shifted covariances with positive lags
+
+    Covariance matrix C_0 is then given by
+
+    C_0 = C[0] - \sum_{i=0}^{p-1} A[i]C[i+1]
+
+    where p is the orde of the autoregressive model.
+
+    """
+
+    yule_walker_matrix, lag_covariances = \
+        _yule_walker_matrix(signals, dimension, order)
+
+    positive_lag_covariances = np.reshape(lag_covariances[1:],
+                                          (dimension*order, dimension))
+
+    lstsq_coeffs = \
+        np.linalg.lstsq(yule_walker_matrix, positive_lag_covariances)[0]
+
+    coeffs = []
+    for index in range(order):
+        coeffs.append(lstsq_coeffs[index*dimension:(index+1)*dimension, ].T)
+
+    coeffs = np.stack(coeffs)
+
+    cov_matrix = np.copy(lag_covariances[0])
+    for i in range(order):
+        cov_matrix -= np.matmul(coeffs[i], lag_covariances[i+1])
+
+    return coeffs, cov_matrix
+
+
+def _optimal_vector_arm(signals, dimension, max_order,
+                        information_criterion='bic'):
+    """
+    Determine optimal auto regressive model by choosing optimal order via
+    Information Criterion
+    Parameters
+    ----------
+    signals : np.ndarray
+        time series data
+    dimension : int
+        dimensionality of the data
+    max_order : int
+        maximal order to consider
+    information_criterion : string
+        bic for Bayesian information_criterion,
+        aic for Akaike information criterion
+
+    Returns
+    -------
+    optimal_coeffs: np.ndarray
+        coefficients of the autoregressive model
+    optimal_cov_mat : np.ndarray
+        covariance matrix of
+    optimal_order : int
+        optimal order
+    """
+
+    length = np.size(signals[0])
+
+    optimal_ic = np.infty
+    optimal_order = 1
+    optimal_coeffs = np.zeros((dimension, dimension, optimal_order))
+    optimal_cov_matrix = np.zeros((dimension, dimension))
+
+    if information_criterion == 'bic':
+        evaluate_ic = bic
+    elif information_criterion == 'aic':
+        evaluate_ic = aic
+    else:
+        raise ValueError(
+            f"Information criterion {information_criterion} not valid")
```

Comment:
it's easily solved by passing `information_criterion=bic` meaning that information_criterion is a callable and not a string. Then you don't need this checking block at all.

Reply:
besides, we're still with py2 and py3.5 that don't know about nice in-place string formatting of a form `f"{}"`

Reply:
Addressed in commit 934b047